src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.cs

```diff
@@ -1412,28 +1412,31 @@ public ValueTask HandleAsync(ProcessAuthenticationContext context)
                 // Yandex returns the email address as a custom "default_email" node:
                 ProviderTypes.Yandex => (string?) context.UserInfoResponse?["default_email"],
 
+                // Genesys returns the email address as a custom "default_email" node:
+                ProviderTypes.Genesys => context.MergedPrincipal.GetClaim("email"),
```

Comment:
Multiple issues here:
  - The code doesn't match the associated comment.
  - You're not supposed to use the merged principal when looking for claims, that should be retrieved from userinfo or token responses.
  - Assuming the email addresses is returned as an `email` node in the userinfo response, you can remove that because that's the standard claim so custom logic isn't needed.